### DISCUSSION TITLE
Num Heads & Num Medusa Layers

### DISCUSSION THREAD

**Discussion by santhosh97** [FasterDecoding/Medusa#44] [General]
Hey All!

I had  of questions regarding Medusa. Has your team tested / experimented / ablation studies with increasing the number of medusa heads / medusa layers? For instance, the default right now is 3 heads and 1 medusa layer, if we changed that to say 6 and 2, do you foresee an even more increase in generation efficiency (say 3x to 4x) with a drawback in accuracy / quality score?

Best,
Santhosh Subramanian

**Comment by leeyeehoo** [comment]
We report the train [Accuracy](https://github.com/FasterDecoding/Medusa/blob/main/assets/medusa_acc.csv) of 5 Medusa heads for Vicuna.

The efficiency in generation is affected by two elements: the acceleration ratio and overhead. Increasing the number of prediction paths on the Medusa tree can enhance the acceleration ratio on one side. Conversely, elongating the input length (comparing extended token k to 1) incurs notable overhead, especially when k exceeds 64, corresponding to the GPU warp of A100—this may vary with other devices. Therefore, you can refer to the Appendix of our blogs to view a plot displaying most of the optimal wall time configurations, which are centered around 64.

**Reply by itayhubara** [reply]
Thank you for this great paper. I wonder what is the acceptance rate for Medusa-2/1 without tree-attention. For Medusa-1 it reduced the speed up by 20-30% (table 3) but no report of acceptance rate. Do you happen to have those numbers? Do you think that Medusa-2 will be less affected by removing the tree attention?
